### DISCUSSION TITLE
How to get data from OAuth Providers?

### DISCUSSION THREAD

**Discussion by zdanl** [appwrite/appwrite#5053] [Q&A] [answered]
### 🔖 Feature description

There is not much information of the OAuth transaction left in the `account.get()` Response object using the Appwrite SDK, and we would need not just the name, standardized per Appwrite Auth, but also the Github URL, API Token, Profile Picture, and such. Is there currently any way to do this, without patching AppWrite?

### 🎤 Pitch

There is many reasons to expose that data to the browser, for ease of integration with other platforms. Please correct me if there already is a way.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by zdanl** [comment]
@eldadfux

**Comment by stnguyen90** [comment]
@zdanl thanks for creating this issue! 🙏🏼 It would be great if this could be split into 3 separate issues so they can be discussed and upvoted accordingly.

**Comment by zdanl** [comment]
> @zdanl thanks for creating this issue! 🙏🏼 It would be great if this could be split into 3 separate issues so they can be discussed and upvoted accordingly.

I have a strong feeling it's gonna be ignored, no matter how much effort I put in. 

Does account.createOAuth2Session() https://appwrite.io/docs/client/account?sdk=web-default#accountCreateOAuth2Session currently have a return object with api key or data from selected scopes? 

If not, is there any advise on the patch beyond ````git clone appwrite & grep "createOAuth2"````?

-D

**Comment by stnguyen90** [comment, marked as answer]
@zdanl yes, it is possible to fetch data from the OAuth provider, assuming adequate scopes were requested. Currently, OAuth data is in the [`Session`](https://appwrite.io/docs/models/session) rather than `Account`. As such, instead of calling `account.get()`, you would call [`account.getSession('current')`](https://appwrite.io/docs/client/account?sdk=web-default#accountGetSession) to get the current `Session`. Assuming the session is an OAuth session, `providerAccessToken` would be populated with the access token from the OAuth provider. You can, then, use that access token to make API calls to the OAuth2 provider.
